tests/mpmc_channel.rs

```diff
@@ -918,6 +941,41 @@ macro_rules! gen_mpmc_tests {
                 assert_send_done(cx, &mut send_fut2, Ok(()));
                 assert_send_done(cx, &mut send_fut3, Ok(()));
             }
+
+            #[test]
+            fn stream_smoke_test() {
+                let channel = ChannelType::new();
+                let (waker, count) = new_count_waker();
+                let cx = &mut Context::from_waker(&waker);
+
+                let stream = channel.stream();
+                pin_mut!(stream);
+
+                assert!(stream.as_mut().poll_next(cx).is_pending());
```

Comment:
Maybe let's add another test which checks for the "immediately ready" variant (send before receive).